diff --git a/targets/ramips-mt7621 b/targets/ramips-mt7621
index 759f96c7..0784ec8e 100644
--- a/targets/ramips-mt7621
+++ b/targets/ramips-mt7621
@@ -170,6 +170,10 @@ device('zyxel-wsm20', 'zyxel_wsm20', {
 
 -- Ubiquiti
 
+device('ubiquiti-usw-flex', 'ubnt_usw-flex', {
+        factory = false,
+})
+
 device('ubiquiti-edgerouter-x', 'ubnt_edgerouter-x', {
 	factory = false,
 	packages = {'-hostapd-mini'},
