diff --git a/package/gluon-core/luasrc/lib/gluon/upgrade/021-interface-roles b/package/gluon-core/luasrc/lib/gluon/upgrade/021-interface-roles
index 6bceb36e..829c12e6 100755
--- a/package/gluon-core/luasrc/lib/gluon/upgrade/021-interface-roles
+++ b/package/gluon-core/luasrc/lib/gluon/upgrade/021-interface-roles
@@ -18,6 +18,16 @@ roles.single = site.interfaces.single.default_roles({unpack(roles.wan)})
 -- was previously enabled
 local mesh_lan_disabled = uci:get('network_gluon-old', 'mesh_lan', 'disabled')
 local mesh_wan_disabled = uci:get('network_gluon-old', 'mesh_wan', 'disabled')
+local mesh_vpn_enabled = uci:get('network_gluon-old', 'mesh_vpn', 'enabled')
+
+--- dealing with wan/uplink
+if mesh_vpn_enabled == '1' then
+ 	util.add_to_set(roles.wan, 'uplink')
+	util.add_to_set(roles.single, 'uplink')
+elseif mesh_vpn_enabled == '0'
+	util.remove_from_set(roles.wan, 'uplink')
+	util.remove_from_set(roles.single, 'uplink')
+end
 if mesh_wan_disabled == '0' then
 	util.add_to_set(roles.wan, 'mesh')
 	util.add_to_set(roles.single, 'mesh')
@@ -25,12 +35,19 @@ elseif mesh_wan_disabled == '1' then
 	util.remove_from_set(roles.wan, 'mesh')
 	util.remove_from_set(roles.single, 'mesh')
 end
+--- WAN/single not vpn nor mesh -> client
+if mesh_vpn_enabled == '0' and mesh_wan_disabled == '1' then
+ 	util.add_to_set(roles.wan, 'client')
+	util.add_to_set(roles.single, 'client')
+end
+
+--- dealing with LAN-function (can not be vpn uplink in 2021.x, but either mesh or client)
 if mesh_lan_disabled == '0' then
+	util.remove_from_set(roles.lan, 'client')
 	util.add_to_set(roles.lan, 'mesh')
-	util.add_to_set(roles.single, 'mesh')
 elseif mesh_lan_disabled == '1' then
 	util.remove_from_set(roles.lan, 'mesh')
-	util.remove_from_set(roles.single, 'mesh')
+	util.add_to_set(roles.lan, 'client')
 end
 
 -- Migration of single to WAN/LAN or vice-versa (an interface was added or removed)
